crowdsourcing/viewsets/task.py

```diff
@@ -227,6 +236,15 @@ def submit_results(self, request, *args, **kwargs):
             task_worker.task_status = task_status
             task_worker.save()
 
+            try:
+                user_preferences, created = models.UserPreferences.objects.get_or_create(user=request.user)
```

Comment:
do not update user preferences in this viewset, use the userpreferences viewset for that